### ISSUE TITLE
Order row click in admin should respect button elements too

### ISSUE THREAD

**Issue by jonathan-dejong** [woocommerce/woocommerce#21955]
**Is your feature request related to a problem? Please describe.**
We often create custom actions for our clients in the order list screen. These are usually AJAX driven because the nature of working directly from the list is to not be redirected or lose your scrolling position. The new "click anywhere" feature of the order row item however makes that impossible if using the `<button>` element. It respects `<a>` but not `<button>`. And in such cases the more semantically correct element is `<button>`. 

**Describe the solution you'd like**
Add a check to https://github.com/woocommerce/woocommerce/blob/d8921de735eb22053394e266bcc04c6227bb8058/assets/js/admin/wc-orders.js#L21 so it also respects clicking on a button element and does not send the user to the edit screen. 

**Describe alternatives you've considered**
I guess the alternative is to always use `<a>` but we shouldn't have to sacrifice proper semantics and better accessibility over something so trivial.

**Comment by willgorham** [comment]
I think there's a `no-link` helper class you can use to prevent this from happening. This PR info might be helpful: https://github.com/woocommerce/woocommerce/pull/18708. Filtering out generic `button`s might be smart too though.

**Comment by kloon** [comment]
See #21966 for fix. Alternatively, you can also just use the .no-link class on the button.

**Comment by jonathan-dejong** [comment]
Thanks for the headsup on that!

I spent a good while trying to figure out any information on this via google and came up with nothing until I actually checked the JS file. So in my opinion the documentation for that solution is poor. 

And honestly the class is an alternative solution as you write but it still does not invalidate this issue. 
It makes perfect sense to exclude generic `button` clicks the same as `a` clicks.
